libs/ml/include/ml/dataloaders/word2vec_loaders/vocab.hpp

```diff
@@ -42,6 +57,101 @@ class Vocab
   SizeType    IndexFromWord(std::string const &word) const;
 };
 
+Vocab::Vocab()
```

Comment:
I would say that an empty constructor is probably the one exemption from the rule of thumb that you should define methods in the class but define them outside the class.
You can ditch this and just add an {} to the definition in the class (or a = default)